cl/recap/mergers.py

```diff
@@ -1736,3 +1739,40 @@ async def process_orphan_documents(
             # exceptions that were previously raised for the
             # processing queue items a second time.
             pass
+
+
+@retry(IntegrityError, tries=3, delay=0.25, backoff=1)
```

Comment:
Yeah, this method, `process_case_query_report`, is similar to `save_iquery_to_docket`. However, due to their different retry approaches, I thought it was better to create a new method.

`process_case_query_report` is executed within the probe task. If there is a race condition, an `IntegrityError` can be raised when saving the docket. In this case, the method for that docket is independently retried using the `@retry` decorator, instead of retrying the whole probe task, which could lead to re-processing previously stored cases .

The `save_iquery_to_docket` method is called by the `make_docket_by_iquery_base` task. Its retry method is based on Celery retries, so the entire task is retried.